packages/h5wasm/src/guards.ts

```diff
@@ -51,7 +51,7 @@ export function isNumericMetadata(
 }
 
 export function isStringMetadata(metadata: Metadata) {
-  return metadata.type === 2;
+  return metadata.type === 3;
```

Comment:
String datasets were given the type `Unknown` as I made a mistake when reading the [type enum](https://github.com/usnistgov/h5wasm/blob/fb407d0c7b2d366485832910d3b43ce70ae6d15e/src/hdf5_util_helpers.d.ts#L10)...

Whoops !